File: codesignal/arcade/2.core/11.spring of integration/1.arrayConversion.cpp
// Given an array of 2k integers (for some integer k), perform the following operations until the 
// array contains only one element:

// On the 1st, 3rd, 5th, etc. iterations (1-based) replace each pair of consecutive elements with their sum;
// On the 2nd, 4th, 6th, etc. iterations replace each pair of consecutive elements with their product.
// After the algorithm has finished, there will be a single element left in the array. Return that element.

#include <iostream>
#include <vector>

class Solution {
public:
  int arrayConversion(std::vector<int> inputArray) {
    int i = 0;
    while (inputArray.size() > 1) {
      std::vector<int> temp;
      if (i % 2 == 0) {
        for (int j = 0; j < inputArray.size(); j += 2) {
          temp.push_back(inputArray[j] + inputArray[j + 1]);
        }
      }
      else {
        for (int j = 0; j < inputArray.size(); j += 2) {
          temp.push_back(inputArray[j] * inputArray[j + 1]);
        }
      }
      inputArray = temp;
      i++;
    }
    return inputArray[0];
  }
};

File: codesignal/arcade/2.core/11.spring of integration/1.arrayConversion.py
# Given an array of 2k integers (for some integer k), perform the following operations until the 
# array contains only one element:

# On the 1st, 3rd, 5th, etc. iterations (1-based) replace each pair of consecutive elements with their sum;
# On the 2nd, 4th, 6th, etc. iterations replace each pair of consecutive elements with their product.
# After the algorithm has finished, there will be a single element left in the array. Return that element.

class Solution:
    def arrayConversion(self, inputArray):
        isOdd = True
        while len(inputArray) != 1:
            if isOdd:
                inputArray = [inputArray[i] + inputArray[i+1] for i in range(0, len(inputArray), 2)]
            else:
                inputArray = [inputArray[i] * inputArray[i+1] for i in range(0, len(inputArray), 2)]
            isOdd = not isOdd
        return inputArray[0]